src/server/test/web/readingsBarGroupQuantity.js

```diff
@@ -392,8 +392,123 @@ mocha.describe('readings API', () => {
+                        {
+                            // u12
+                            name: 'kg CO₂',
+                            identifier: '',
+                            unitRepresent: Unit.unitRepresentType.QUANTITY,
+                            secInRate: 3600,
+                            typeOfUnit: Unit.unitType.UNIT,
+                            suffix: 'CO₂',
+                            displayable: Unit.displayableType.ALL,
+                            preferredDisplay: false,
+                            note: 'special unit'
+                        },
+                        {
+                            // u13
+                            name: 'pound',
+                            identifier: 'lb',
+                            unitRepresent: Unit.unitRepresentType.QUANTITY,
+                            secInRate: 3600,
+                            typeOfUnit: Unit.unitType.UNIT,
+                            suffix: '',
+                            displayable: Unit.displayableType.ALL,
+                            preferredDisplay: false,
+                            note: 'special unit'
+                        }
+                    ];
+                    const conversionData = [
+                        {
+                            // c11
+                            sourceName: 'Electric_Utility',
+                            destinationName: 'kg CO₂',
+                            bidirectional: false,
+                            slope: 0.709,
+                            intercept: 0,
+                            note: 'Electric_Utility → kg CO₂'
+                        },
+                        {
+                            // c12
+                            sourceName: 'kg CO₂',
+                            destinationName: 'kg',
+                            bidirectional: false,
+                            slope: 1,
+                            intercept: 0,
+                            note: 'CO₂ → kg'
+                        },
+                        {
+                            // c13
+                            sourceName: 'kg',
+                            destinationName: 'metric ton',
+                            bidirectional: true,
+                            slope: 1e-3,
+                            intercept: 0,
+                            note: 'kg → Metric ton'
+                        },
+                        {
+                            // c14 
+                            sourceName: 'pound',
+                            destinationName: 'metric ton',
+                            bidirectional: true,
+                            slope: 454.545454,
+                            intercept: 0,
+                            note: 'lbs → metric tons'
+                        }
+                    ];
+                    
+                    //loads data into database
+                    await prepareTest(unitDatakWh, conversionDatakWh, meterDatakWhGroups, groupDatakWh);
```

Comment:
The issue is that the provided meters and groups have a default graphic unit of kWh but this test does not create that unit. This causes the code to fail. What needs to happen is the needed meters and groups must be created within this test. It is similar to the standard ones but modified somewhat. I created the needed code to test the solution. If you want that code or for me to commit/push the solution then let me know.